common/lib/xmodule/xmodule/js/src/problem/edit.coffee

```diff
@@ -570,12 +576,83 @@ class @MarkdownEditingDescriptor extends XModule.Descriptor
 
       // if we've come across demand hints, wrap in <demandhint> at the end
       if (demandhints) {
-          demandhints = '\n<demandhint>\n' + demandhints + '</demandhint>';
+          demandHintTags.push(demandhints);
       }
 
-      // make all elements descendants of a single problem element
-      xml = '<problem>\n' + xml + demandhints + '\n</problem>';
+      // make selector to search responsetypes in xml
+      var responseTypesSelector = responseTypes.join(', ');
+
+      // these will be placed at outside the end of responsetype
+      var independentTagNames = ['solution'];
```

Comment:
@muzaffaryousaf This will happen in [TNL-4822](https://openedx.atlassian.net/browse/TNL-4822)